### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by nodar-chkuaselidze**
- minor refactors.
- minor proposal bug.
- client - fix joinWallet/join methods conflict.
- update dependencies
- parse multisig and root paths from options (https://github.com/bcoin-org/bmultisig/issues/15)
  - `path` will always be assumed as wallet path.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ docs/
 node_modules/
 npm-debug.log
 coverage/
+package-lock.json
```
Comment (tynes) [inline .gitignore]: Why are you ignoring the `package-lock.json`? Isn't is useful to check into the repo to have the exact state of `node_modules` defined?

File: lib/client.js [hunk 2]
```diff
@@ -6,18 +6,77 @@
...
+    if (options.multisigPath != null) {
+      enforce(typeof options.multisigPath === 'string',
+        'multisigPath', 'string');
+
+      this.multisigPath = options.multisigPath;
+      this.path = options.multisigPath;
+    }
+
+    if (options.multisigPath != null && options.path == null) {
+      const up = this.multisigPath.slice(-1) === '/' ? '../' : '/../';
```
Comment (tynes) [inline lib/client.js]: Can you add a comment here?

File: test/http-test.js [hunk 3]
```diff
@@ -619,6 +619,61 @@ describe('HTTP', function () {
...
+      const multisigPath = '/test/multisig';
+
+      const client = new MultisigClient({ path });
+
+      assert.strictEqual(client.rootPath, path);
+      assert.strictEqual(client.path, multisigPath);
+      assert.strictEqual(client.multisigPath, multisigPath);
+    });
+
+    it('should parse root path from multisig path', () => {
```
Comment (tynes) [inline test/http-test.js]: Maybe specify the difference between these 2 test cases, they are different based on the trailing `/` but the description of both is exactly the same

File: .gitignore (hunk 1 above)
Comment (nodar-chkuaselidze) [reply .gitignore]: 👍 I was thinking of removing it but then decided to leave it there, forgot to remove .gitignore.

**Comment by nodar-chkuaselidze** [comment]
After major things are done I will update README.

**Pull request merged**